vms/platformvm/utxo/camino_locked.go

```diff
@@ -340,106 +377,100 @@ func (h *handler) Lock(
 			})
 			signers = append(signers, inSigners)
 
-			ownerID, err := GetOwnerID(out)
-			if err != nil {
-				// We couldn't get owner of this output, so move on to the next one
-				continue
+			otherLockTxID := lockIDs.DepositTxID
+			if appliedLockState == locked.StateDeposited {
+				otherLockTxID = lockIDs.BondTxID
 			}
 
-			ownerAmounts, ok := insAmounts[ownerID]
+			ownerAmounts, ok := insAmounts[*lockedOwnerID.ownersID]
 			if !ok {
 				ownerAmounts = OwnerAmounts{
 					amounts: make(map[ids.ID]lockedAndRemainedAmounts),
-					owners:  innerOut.OutputOwners,
-					signers: inSigners,
+					owners:  *lockedOwnerID.owners,
 				}
 			}
 
-			otherLockTxID := lockIDs.DepositTxID
-			if appliedLockState == locked.StateDeposited {
-				otherLockTxID = lockIDs.BondTxID
-			}
-
 			amounts := ownerAmounts.amounts[otherLockTxID]
-
 			newAmount, err := math.Add64(amounts.locked, amountToLock)
 			if err != nil {
-				// We couldn't sum this input's amount, so move on to the next one
-				continue
+				return nil, nil, nil, err
```

Comment:
why error instead of continue here?

Reply:
Because you already inserted stuff in internal structures